labelbox/data/annotation_types/collection.py

```diff
@@ -179,22 +179,14 @@ def _apply_threaded(self, fns, max_concurrency, *args):
 
 class LabelGenerator(PrefetchGenerator):
     """
-    A container for interacting with a collection of labels.
-
-    Use this class if you have larger data. It is slightly harder to work with
-    than the LabelList but will be much more memory efficient.
+    A container for interacting with a large collection of labels. 
+    For a small number of labels, just use a list of Label objects.
     """
 
     def __init__(self, data: Generator[Label, None, None], *args, **kwargs):
         self._fns = {}
         super().__init__(data, *args, **kwargs)
 
-    def as_list(self) -> "LabelList":
```

Comment:
Shoudn't we wait till the 4.0 release to make a breaking change? Until then we could mention that this will be deprecated in the docstring.

Reply:
This is why I am doing this PR :) should I revert this change? cc @kkim-labelbox

Reply:
However, keep in mind, I have also removed all usages of this method both in the sdk codebase and Notebooks

Reply:
I think it's sufficient to leave the warning here, and don't remove the method, since it is a breaking change. As long as we remove usage of LabelList and `as_list` in our codebase, that should satisfy our deprecation requirements

Reply:
@whistler restored